chore: Add a separate E2E tests job for the ASM module in 'ci.yaml' and 'ci-merge-check.yaml'

luoyifeng11 opened the pull request:
The primary objective is to resolve the failures in ASM E2E tests. For a comprehensive analysis of the reasons and the goals of the adjustments, please refer to the ticket: [CXSPA-3009](https://jira.tools.sap/browse/CXSPA-3009)

luoyifeng11 committed:
test: test e2e for asm

luoyifeng11 committed:
test: modify workflow

luoyifeng11 committed:
test: modify workflow

luoyifeng11 committed:
test: modify workflow

luoyifeng11 committed:
test: modify workflow

luoyifeng11 committed:
test: modify workflow

luoyifeng11 committed:
test: modify workflow

luoyifeng11 committed:
test: modify workflow

luoyifeng11 committed:
test: modify workflow

luoyifeng11 committed:
test: modify workflow

luoyifeng11 committed:
CXSPA-3009: add asm job in workflow

luoyifeng11 committed:
CXSPA-3009: Delete test code

luoyifeng11 committed:
Merge branch 'develop-6.1.x' into bugfix/CXSPA-3009

luoyifeng11 committed:
Merge branch 'develop-6.1.x' into bugfix/CXSPA-3009

luoyifeng11 committed:
CXSPA-3009: Delete test code

luoyifeng11 committed:
CXSPA-3009: Adjustment script

luoyifeng11 committed:
CXSPA-3009: modify agent user for b2b test

luoyifeng11 committed:
CXSPA-3009: rename file

luoyifeng11 committed:
CXSPA-3009: rename file

luoyifeng11 committed:
CXSPA-3009: rename param

luoyifeng11 committed:
CXSPA-3009: refine param tag

luoyifeng11 committed:
CXSPA-3009: refine -group

luoyifeng11 committed:
CXSPA-3009: Adjustment script

luoyifeng11 committed:
CXSPA-3009: Adjustment script

luoyifeng11 committed:
CXSPA-3009: testing

luoyifeng11 committed:
CXSPA-3009: testing

luoyifeng11 committed:
CXSPA-3009: testing

luoyifeng11 committed:
CXSPA-3009: Delete test code

luoyifeng11 committed:
Merge branch 'develop-6.1.x' into bugfix/CXSPA-3009

niehuayang reviewed (dismissed):
Looks good to me, Please do not merge for now, there may be experts who want to review:)

niehuayang commented on projects/storefrontapp-e2e-cypress/package.json:
```diff
@@ -30,7 +30,11 @@
     "cy:run:ci:digital-payments": "cypress run --config-file cypress.config.ci.ts --env API_URL=https://backoffice.cp96avkh5f-sapcxteam1-d5-public.model-t.cc.commerce.ondemand.com --record --key $CYPRESS_KEY --tag \"2105,digital-payments\" --group DIGITAL-PAYMENTS --spec \"cypress/e2e/vendor/digital-payments/*.e2e.cy.ts\"",
-    "cy:run:ci:epd-visualization": "cypress run --config-file cypress.config.ci.ts --env API_URL=https://api.cp96avkh5f-integrati1-d1-public.model-t.cc.commerce.ondemand.com,BASE_SITE=powertools-epdvisualization-spa --record --key $CYPRESS_KEY --tag \"2105,epd-visualization\" --group EPD_VISUALIZATION --ci-build-id $BUILD_NUMBER --spec \"cypress/e2e/vendor/epd-visualization/**/*.e2e.cy.ts\""
+    "cy:run:ci:epd-visualization": "cypress run --config-file cypress.config.ci.ts --env API_URL=https://api.cp96avkh5f-integrati1-d1-public.model-t.cc.commerce.ondemand.com,BASE_SITE=powertools-epdvisualization-spa --record --key $CYPRESS_KEY --tag \"2105,epd-visualization\" --group EPD_VISUALIZATION --ci-build-id $BUILD_NUMBER --spec \"cypress/e2e/vendor/epd-visualization/**/*.e2e.cy.ts\"",
+    "cy:run:ci:asm": "cypress run --config-file cypress.config.ci.ts --record --key $CYPRESS_KEY --tag \"b2c,asm,all\" --parallel --ci-build-id $BUILD_NUMBER --group ASM_B2C --spec \"cypress/e2e/vendor/asm/b2c/**/*e2e.cy.ts\" --reporter junit --reporter-options mochaFile=results/e2e-test-b2c-asm-[hash].xml",
```
I noticed that the new container did not include ASM accessibility related e2e tests. They are still in the original container, so they are still parallel, which maybe cause some issues. Could yoiu please also add ASM accessibility  related e2e tests to the new container?

luoyifeng11 replied:
Oh, I overlooking that.
I have indeed created separate "accessibility" directories within the "b2c" and "b2b" directories under the "vendor" folder. These directories will contain the accessibility tests specific to the respective B2C and B2B. These tests will be executed within the "asm" job.

luoyifeng11 committed:
CXSPA-3009: asm accessibility tests are moved to the vender directory

luoyifeng11 committed:
Merge branch 'develop-6.1.x' into bugfix/CXSPA-3009

Larisa-Staroverova committed:
fix: Review feedback - add quantity stepper next to add-to-cart button (#17476)

Review feedback: add quantity stepper next to add-to-cart button

Closes CXSPA-3193

Larisa-Staroverova committed:
fix: Failing cypress test: quantity stepper next to add-to-cart button (#17477)

Failing cypress test: quantity stepper next to add-to-cart button

Closes CXSPA-3193

Larisa-Staroverova committed:
fix: Set z-index property to a higher number for navigation header (#17475)

Set z-index property to a higher number for the navigation header to hide completely all elements behind the header when it is expanded

Closes CXSPA-3540

KateChuen committed:
CCV2 P5 Spartacus deploy test with webpack 5.76.0 (#17470)

Co-authored-by: Giancarlo Cordero Ortiz <46171897+giancorderoortiz@users.noreply.github.com>

luoyifeng11 committed:
Merge remote-tracking branch 'origin/develop-6.2.x' into bugfix/CXSPA-3009
